rpcs3/Emu/Cell/Modules/cellCamera.cpp

```diff
@@ -454,30 +390,24 @@ s32 cellCameraOpenEx(s32 dev_num, vm::ptr<CellCameraInfoEx> info)
 
 	const auto g_camera = fxm::get<camera_thread>();
 
-	if (!g_camera)
-	{
-		return CELL_CAMERA_ERROR_NOT_INIT;
-	}
-
-	s32 status = set_and_send_read_mode(dev_num, info->read_mode);
-	if (status != CELL_OK)
+	s32 status;
+	if ((status = cellCameraSetAttribute(dev_num, CELL_CAMERA_READMODE, info->read_mode, 0) != CELL_OK))
 	{
 		return status;
 	}
-
-	status = cellCameraSetAttribute(dev_num, CELL_CAMERA_GAMEPID, 0, 0); // yup, that's what libGem does
-	if (status != CELL_OK)
+	if (info->read_mode == CELL_CAMERA_READ_DIRECT)
 	{
-		return status;
+		if ((status = cellCameraSetAttribute(dev_num, CELL_CAMERA_GAMEPID, status, 0) != CELL_OK))
+		{
+			return status;
+		}
 	}
 
 	if (!check_dev_num)
 	{
 		return CELL_CAMERA_ERROR_PARAM;
 	}
 
-	semaphore_lock lock(g_camera->mutex);
-
 	if (g_camera->is_open)
```

Comment:
is_open should be initialized as false along with the other members of camera_thread

Reply:
ugh now we need to connect fake cameras at init I guess.